From 01ba67004fdeaa1c205935ffb3e6739a9005cc21 Mon Sep 17 00:00:00 2001
From: Minkyu Kang <mk7.kang@samsung.com>
Date: Wed, 29 Feb 2012 11:28:03 +0900
Subject: [PATCH] JQM: Change button design to meet TIZEN UI

Signed-off-by: Lee Wongi <wongi11.lee@samsung.com>
Signed-off-by: Minkyu Kang <mk7.kang@samsung.com>
---
 .../js/jquery.mobile.buttonMarkup.js               |   61 +++++++++++++++++++-
 1 files changed, 58 insertions(+), 3 deletions(-)

diff --git a/libs/js/jquery-mobile-1.0.1pre/js/jquery.mobile.buttonMarkup.js b/libs/js/jquery-mobile-1.0.1pre/js/jquery.mobile.buttonMarkup.js
index d192e37..93485f1 100644
--- a/libs/js/jquery-mobile-1.0.1pre/js/jquery.mobile.buttonMarkup.js
+++ b/libs/js/jquery-mobile-1.0.1pre/js/jquery.mobile.buttonMarkup.js
@@ -2,6 +2,21 @@
 * "buttons" plugin - for making button-like links
 */
 
+/*
+ * Button Markup modified for TIZEN style.
+ *
+ * HTML Attributes:
+ *
+ *		data-role: button
+ *		data-style: circle, nobg, edit
+ *
+ * Examples:
+ *
+ *  	<div data-role="button" data-inline="true" data-icon="send" data-style="circle"></div>
+ *  	<div data-role="button" data-inline="true" data-icon="favorite" data-style="nobg">
+ *  	<div data-role="button" data-inline="true" data-icon="editminus" data-style="edit"></div>
+ */
+
 ( function( $, undefined ) {
 
 $.fn.buttonMarkup = function( options ) {
@@ -73,8 +88,48 @@ $.fn.buttonMarkup = function( options ) {
 			buttonClass += " ui-shadow";
 		}
 
-		e.setAttribute( "data-" + $.mobile.ns + "theme", o.theme );
-		el.addClass( buttonClass );
+		/* TIZEN style markup */
+		buttonStyle = el.jqmData("style");
+
+		if ( buttonStyle == "circle" ) {
+			/* style : no text, Icon only */
+			buttonClass += " ui-btn-corner-circle";
+			buttonClass += " ui-btn-icon_only";
+		} else if ( buttonStyle == "nobg" ) {
+			/* style : no text, Icon only, no bg */
+			buttonClass += " ui-btn-icon-nobg";
+			buttonClass += " ui-btn-icon_only";
+		} else if ( buttonStyle == "edit" ) {
+			buttonClass += " ui-btn-edit";
+		}
+
+		if ( o.icon ) {
+			if ( $(el).text().length > 0 ) {
+				o.iconpos == "right" ?
+					textClass += " ui-btn-text-padding-right" :
+					textClass += " ui-btn-text-padding-left";
+
+				innerClass += " ui-btn-hastxt";
+			} else {
+				if ( buttonStyle == "circle" ) {
+					/* style : no text, Icon only */
+					innerClass += " ui-btn-corner-circle";
+				} else if ( buttonStyle == "nobg" ) {
+					/* style : no text, Icon only, no bg */
+					innerClass += " ui-btn-icon-nobg";
+				}
+
+				buttonClass += " ui-btn-icon_only";
+				innerClass += " ui-btn-icon-only";
+			}
+		} else {
+			if ( $(el).text().length > 0 ) {
+				innerClass += " ui-btn-hastxt";
+			}
+		}
+
+		el.attr( "data-" + $.mobile.ns + "theme", o.theme )
+			.addClass( buttonClass );
 
 		buttonInner.className = innerClass;
 		buttonInner.setAttribute("aria-hidden", "true");
@@ -96,7 +151,7 @@ $.fn.buttonMarkup = function( options ) {
 		// TODO obviously it would be nice to pull this element out instead of
 		// retrieving it from the DOM again, but this change is much less obtrusive
 		// and 1.0 draws nigh
-		$.data( e, 'textWrapper', $( buttonText ) );
+		el.data( 'textWrapper', $( buttonText ) );
 	}
 
 	return this;
-- 
1.7.5.4

