From 88c84e8e486540403b51573a28350a707f4b60ef Mon Sep 17 00:00:00 2001
From: Minkyu Kang <mk7.kang@samsung.com>
Date: Wed, 29 Feb 2012 16:56:24 +0900
Subject: [PATCH] JQM change input's type on fixed header

Signed-off-by: Lee Wongi <wongi11.lee@samsung.com>
---
 .../js/jquery.mobile.fixHeaderFooter.js            |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libs/js/jquery-mobile-1.0.1pre/js/jquery.mobile.fixHeaderFooter.js b/libs/js/jquery-mobile-1.0.1pre/js/jquery.mobile.fixHeaderFooter.js
index 0216c65..8fd9357 100644
--- a/libs/js/jquery-mobile-1.0.1pre/js/jquery.mobile.fixHeaderFooter.js
+++ b/libs/js/jquery-mobile-1.0.1pre/js/jquery.mobile.fixHeaderFooter.js
@@ -291,7 +291,7 @@ $.mobile.fixedToolbars = (function() {
 							.addClass("ui-title-content-option-header-expanded-1line-height");
 					}
 				}
-			} else if ( s_theme_header.find("input").jqmData("type") == "search" ) {
+			} else if( s_theme_header.find("input").attr("type") === "search" || s_theme_header.find("input").attr("type") === "tizen-search" ) {
 				s_theme_content
 					.removeClass("ui-title-content-" + title_style + "-height")
 					.addClass("ui-title-content-search");
-- 
1.7.5.4

