From bf9dc02776446faee9e5587360584d9d9b9b135e Mon Sep 17 00:00:00 2001
From: Youmin Ha <youmin.ha@samsung.com>
Date: Tue, 3 Jul 2012 15:07:07 +0900
Subject: [PATCH] JQM:Prevent blinking on page transition

Signed-off-by: Minkyu Kang <mk7.kang@samsung.com>
Signed-off-by: Youmin Ha <youmin.ha@samsung.com>
---
 .../js/jquery.mobile.transition.js                 |   11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/libs/js/jquery-mobile-1.1.0/js/jquery.mobile.transition.js b/libs/js/jquery-mobile-1.1.0/js/jquery.mobile.transition.js
index 56f93a6..e5555e8 100644
--- a/libs/js/jquery-mobile-1.1.0/js/jquery.mobile.transition.js
+++ b/libs/js/jquery-mobile-1.1.0/js/jquery.mobile.transition.js
@@ -29,6 +29,15 @@ var createHandler = function( sequential ){
 				$.mobile.pageContainer.toggleClass( "ui-mobile-viewport-transitioning viewport-" + name );
 			},
 			scrollPage = function(){
+				// Prevent blinking on page scrolling in Tizen/Android devices.
+				// Don't scoll window, when current scroll top(scrollTop()) is already at toScroll,
+				// or when current scroll top is 0 and toScroll is same to defaultHomeScroll
+				// (which means the top position of page). In these case, page scrolling is not needed.
+				var st = $( window ).scrollTop();
+				if( st === toScroll || ( $.mobile.defaultHomeScroll === toScroll && st == 0 ) ) {
+					return;
+				}
+
 				// By using scrollTo instead of silentScroll, we can keep things better in order
 				// Just to be precautios, disable scrollstart listening like silentScroll would
 				$.event.special.scrollstart.enabled = false;
@@ -150,4 +159,4 @@ $.mobile.transitionFallbacks = {};
 })( jQuery, this );
 //>>excludeStart("jqmBuildExclude", pragmas.jqmBuildExclude);
 });
-//>>excludeEnd("jqmBuildExclude");
\ No newline at end of file
+//>>excludeEnd("jqmBuildExclude");
-- 
1.7.9.5

