From 8692febd91d64010547411d1706eb72fc1fe9601 Mon Sep 17 00:00:00 2001
From: Youmin Ha <youmin.ha@samsung.com>
Date: Mon, 26 Nov 2012 22:00:38 +0900
Subject: [PATCH] JQM:nolabel n favorite class for check support

Signed-off-by: Koeun Choi <koeun.choi@samsung.com>
Signed-off-by: Youmin Ha <youmin.ha@samsung.com>
---
 .../js/widgets/forms/checkboxradio.js              |   12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/libs/js/jquery-mobile-1.2.0/js/widgets/forms/checkboxradio.js b/libs/js/jquery-mobile-1.2.0/js/widgets/forms/checkboxradio.js
index 2e1114f..4ee4523 100644
--- a/libs/js/jquery-mobile-1.2.0/js/widgets/forms/checkboxradio.js
+++ b/libs/js/jquery-mobile-1.2.0/js/widgets/forms/checkboxradio.js
@@ -44,6 +44,11 @@ $.widget( "mobile.checkboxradio", $.mobile.widget, {
 			return;
 		}
 
+		// Support fake label
+		if ( label.length == 0 && ( input[0].id !== '' ) ) {
+			label = $( "<label for='" + input[0].id + "'></label>" );
+		}
+
 		// Expose for other methods
 		$.extend( this, {
 			label: label,
@@ -71,6 +77,10 @@ $.widget( "mobile.checkboxradio", $.mobile.widget, {
 		var wrapper = document.createElement('div');
 		wrapper.className = 'ui-' + inputtype;
 
+		if ( input.hasClass( "favorite" ) ) {
+			wrapper.className += ' favorite';
+		}
+
 		input.add( label ).wrapAll( wrapper );
 
 		label.bind({
-- 
1.7.9.5

