From 252820a3ea292bf652d2ecd644c0126152a20f46 Mon Sep 17 00:00:00 2001
From: Jun Jinhyuk <jinhyuk.jun@samsung.com>
Date: Wed, 28 Nov 2012 14:25:58 +0900
Subject: [PATCH] JQM add back btn option and remove jqm back btn

Signed-off-by: Jun Jinhyuk <jinhyuk.jun@samsung.com>
---
 .../js/widgets/page.sections.js                    |   16 ++--------------
 1 files changed, 2 insertions(+), 14 deletions(-)

diff --git a/libs/js/jquery-mobile-1.2.0/js/widgets/page.sections.js b/libs/js/jquery-mobile-1.2.0/js/widgets/page.sections.js
index 7c05ecb..cbf6786 100644
--- a/libs/js/jquery-mobile-1.2.0/js/widgets/page.sections.js
+++ b/libs/js/jquery-mobile-1.2.0/js/widgets/page.sections.js
@@ -8,11 +8,12 @@ define( [ "jquery", "./page", "../jquery.mobile.core" ], function( $ ) {
 (function( $, undefined ) {
 
 $.mobile.page.prototype.options.backBtnText  = "Back";
-$.mobile.page.prototype.options.addBackBtn   = false;
+$.mobile.page.prototype.options.addBackBtn   = "footer";
 $.mobile.page.prototype.options.backBtnTheme = null;
 $.mobile.page.prototype.options.headerTheme  = "a";
 $.mobile.page.prototype.options.footerTheme  = "a";
 $.mobile.page.prototype.options.contentTheme = null;
+$.mobile.page.prototype.options.footerExist  = true;
 
 // NOTE bind used to force this binding to run before the buttonMarkup binding
 //      which expects .ui-footer top be applied in its gigantic selector
@@ -61,19 +62,6 @@ $( document ).bind( "pagecreate", function( e ) {
 				rightbtn = rightbtn || $headeranchors.eq( 1 ).addClass( "ui-btn-right" ).length;
 			}
 
-			// Auto-add back btn on pages beyond first view
-			if ( o.addBackBtn &&
-				role === "header" &&
-				$( ".ui-page" ).length > 1 &&
-				$page.jqmData( "url" ) !== $.mobile.path.stripHash( location.hash ) &&
-				!leftbtn ) {
-
-				backBtn = $( "<a href='javascript:void(0);' class='ui-btn-left' data-"+ $.mobile.ns +"rel='back' data-"+ $.mobile.ns +"icon='arrow-l'>"+ o.backBtnText +"</a>" )
-					// If theme is provided, override default inheritance
-					.attr( "data-"+ $.mobile.ns +"theme", o.backBtnTheme || thisTheme )
-					.prependTo( $this );
-			}
-
 			// Page title
 			$this.children( "h1, h2, h3, h4, h5, h6" )
 				.addClass( "ui-title" )
-- 
1.7.4.1

