From 4b9929185d24ea9230e51bf1f02692e7da69a8cc Mon Sep 17 00:00:00 2001
From: Minkyu Kang <mk7.kang@samsung.com>
Date: Thu, 29 Nov 2012 10:41:59 +0900
Subject: [PATCH] JQM: remove search from forms.textinput.

Signed-off-by: wongi11.lee <wongi11.lee@samsung.com>
Signed-off-by: Minkyu Kang <mk7.kang@samsung.com>
---
 .../js/widgets/forms/textinput.js                  |   46 +++-----------------
 1 file changed, 7 insertions(+), 39 deletions(-)

diff --git a/libs/js/jquery-mobile-1.2.0/js/widgets/forms/textinput.js b/libs/js/jquery-mobile-1.2.0/js/widgets/forms/textinput.js
index 08505ed..530cdb8 100644
--- a/libs/js/jquery-mobile-1.2.0/js/widgets/forms/textinput.js
+++ b/libs/js/jquery-mobile-1.2.0/js/widgets/forms/textinput.js
@@ -14,8 +14,7 @@ $.widget( "mobile.textinput", $.mobile.widget, {
 		theme: null,
 		// This option defaults to true on iOS devices.
 		preventFocusZoom: /iPhone|iPad|iPod/.test( navigator.platform ) && navigator.userAgent.indexOf( "AppleWebKit" ) > -1,
-		initSelector: "input[type='text'], input[type='search'], :jqmData(type='search'), input[type='number'], :jqmData(type='number'), input[type='password'], input[type='email'], input[type='url'], input[type='tel'], textarea, input[type='time'], input[type='date'], input[type='month'], input[type='week'], input[type='datetime'], input[type='datetime-local'], input[type='color'], input:not([type])",
-		clearSearchButtonText: "clear text",
+		initSelector: "input[type='text'], input[type='number'], :jqmData(type='number'), input[type='password'], input[type='email'], input[type='url'], input[type='tel'], textarea, input[type='time'], input[type='date'], input[type='month'], input[type='week'], input[type='datetime'], input[type='datetime-local'], input[type='color'], input:not([type])",
 		disabled: false
 	},
 
@@ -54,37 +53,6 @@ $.widget( "mobile.textinput", $.mobile.widget, {
 			input[0].setAttribute( "autocomplete", "off" );
 		}
 
-
-		//"search" input widget
-		if ( input.is( "[type='search'],:jqmData(type='search')" ) ) {
-
-			focusedEl = input.wrap( "<div class='ui-input-search ui-shadow-inset ui-btn-corner-all ui-btn-shadow ui-icon-searchfield" + themeclass + miniclass + "'></div>" ).parent();
-			clearbtn = $( "<a href='#' class='ui-input-clear' title='" + o.clearSearchButtonText + "'>" + o.clearSearchButtonText + "</a>" )
-				.bind('click', function( event ) {
-					input
-						.val( "" )
-						.focus()
-						.trigger( "change" );
-					clearbtn.addClass( "ui-input-clear-hidden" );
-					event.preventDefault();
-				})
-				.appendTo( focusedEl )
-				.buttonMarkup({
-					icon: "delete",
-					iconpos: "notext",
-					corners: true,
-					shadow: true,
-					mini: mini
-				});
-
-			toggleClear();
-
-			input.bind( 'paste cut keyup focus change blur', toggleClear );
-
-		} else {
-			input.addClass( "ui-corner-all ui-shadow-inset" + themeclass + miniclass );
-		}
-
 		input.focus(function() {
 				focusedEl.addClass( $.mobile.focusClass );
 			})
@@ -141,10 +109,10 @@ $.widget( "mobile.textinput", $.mobile.widget, {
 
 	disable: function() {
 		var $el;
-		if ( this.element.attr( "disabled", true ).is( "[type='search'], :jqmData(type='search')" ) ) {
-			$el = this.element.parent();
-		} else {
+		if ( this.element.attr( "disabled", true ) ) {
 			$el = this.element;
+		} else {
+			return;
 		}
 		$el.addClass( "ui-disabled" );
 		return this._setOption( "disabled", true );
@@ -154,10 +122,10 @@ $.widget( "mobile.textinput", $.mobile.widget, {
 		var $el;
 
 		// TODO using more than one line of code is acceptable ;)
-		if ( this.element.attr( "disabled", false ).is( "[type='search'], :jqmData(type='search')" ) ) {
-			$el = this.element.parent();
-		} else {
+		if ( this.element.attr( "disabled", false ) ) {
 			$el = this.element;
+		} else {
+			return;
 		}
 		$el.removeClass( "ui-disabled" );
 		return this._setOption( "disabled", false );
-- 
1.7.9.5

