From 79d7d678ec75404aa8a927ec337d8563ed26e9fc Mon Sep 17 00:00:00 2001
From: Wonseop Kim <wonseop.kim@samsung.com>
Date: Sat, 16 Mar 2013 17:26:01 +0900
Subject: [PATCH] JQM: Support accessibility for button

---
 .../js/jquery.mobile.buttonMarkup.js               |    9 +++++++
 1 file changed, 9 insertions(+)

diff --git a/libs/js/jquery-mobile-1.2.0/js/jquery.mobile.buttonMarkup.js b/libs/js/jquery-mobile-1.2.0/js/jquery.mobile.buttonMarkup.js
index 7c17ba6..cd8e39d 100644
--- a/libs/js/jquery-mobile-1.2.0/js/jquery.mobile.buttonMarkup.js
+++ b/libs/js/jquery-mobile-1.2.0/js/jquery.mobile.buttonMarkup.js
@@ -52,6 +52,11 @@ $.fn.buttonMarkup = function( options ) {
 			e.setAttribute( "aria-owns", e.getAttribute( "href" ) );
 		}
 
+		if ( e.tagName !== "LI" && e.tagName !== "LABEL" ) {
+			e.setAttribute( "role", "button" );
+			e.setAttribute( "tabindex", "0" );
+		}
+
 		// Check if this element is already enhanced
 		buttonElements = $.data( ( ( e.tagName === "INPUT" || e.tagName === "BUTTON" ) ? e.parentNode : e ), "buttonElements" );
 
@@ -131,6 +136,10 @@ $.fn.buttonMarkup = function( options ) {
 
 				buttonClass += " ui-btn-icon_only";
 				innerClass += " ui-btn-icon-only";
+
+				if ( e.tagName !== "LABEL" ) {
+					$( el ).text( o.icon.replace( "naviframe-", "" ) );
+				}
 			}
 		} else {
 			if ( $(el).text().length > 0 ) {
-- 
1.7.9.5

