From 6a2454411d5416ec39c53c5fb9532ca68363d504 Mon Sep 17 00:00:00 2001
From: Wonseop Kim <wonseop.kim@samsung.com>
Date: Wed, 20 Mar 2013 20:44:35 +0900
Subject: [PATCH] JQM: Support accessibility for page-section

---
 .../js/widgets/page.sections.js                    |   21 ++++++++++++++------
 1 file changed, 15 insertions(+), 6 deletions(-)

diff --git a/libs/js/jquery-mobile-1.2.0/js/widgets/page.sections.js b/libs/js/jquery-mobile-1.2.0/js/widgets/page.sections.js
index e00684c..1a7b373 100644
--- a/libs/js/jquery-mobile-1.2.0/js/widgets/page.sections.js
+++ b/libs/js/jquery-mobile-1.2.0/js/widgets/page.sections.js
@@ -61,6 +61,10 @@ $.mobile.$document.bind( "pagecreate", function( e ) {
 				leftbtn = leftbtn || $headeranchors.eq( 0 ).not( ".ui-btn-right" ).addClass( "ui-btn-left" ).length;
 
 				rightbtn = rightbtn || $headeranchors.eq( 1 ).addClass( "ui-btn-right" ).length;
+
+				$( $headeranchors.get().reverse() ).each( function ( i ) {
+					$( this ).addClass( "ui-btn-right-" + i );
+				});
 			}
 
 			// Auto-add back btn on pages beyond first view
@@ -75,12 +79,13 @@ $.mobile.$document.bind( "pagecreate", function( e ) {
 					$dest = $page.find( ".ui-footer" );
 				}
 
-				backBtn = $( "<a href='javascript:void(0);' class='ui-btn-back' data-"+ $.mobile.ns +"rel='back' data-"+ $.mobile.ns +"icon='header-back-btn'></a>" )
-					// If theme is provided, override default inheritance
-					.attr( "data-"+ $.mobile.ns +"theme", o.backBtnTheme || thisTheme );
-
 				if ( !$dest.find( ".ui-btn-back" ).length ) {
-					backBtn.prependTo( $dest );
+					backBtn = $( "<a href='javascript:void(0);' class='ui-btn-back' data-" + $.mobile.ns + "rel='back'></a>" )
+						// // If theme is provided, override default inheritance
+						.buttonMarkup( { icon: "header-back-btn", theme: o.backBtnTheme || thisTheme } );
+
+					backBtn.find( ".ui-btn-text" ).text( o.backBtnText );
+ 					backBtn.appendTo( $dest );
 				}
 			}
 
@@ -90,9 +94,13 @@ $.mobile.$document.bind( "pagecreate", function( e ) {
 				// Regardless of h element number in src, it becomes h1 for the enhanced page
 				.attr({
 					"role": "heading",
-					"aria-level": "1"
+					"aria-level": "1",
+					"aria-label": "title",
+					"tabindex": "0"
 				});
 
+			$( ".ui-title-text-sub" ).attr( { "tabindex": "0", "aria-label": "subtitle" } );
+
 		} else if ( role === "content" ) {
 			if ( contentTheme ) {
 				$this.addClass( "ui-body-" + ( contentTheme ) );
-- 
1.7.9.5

