From f5525377fccf6744ad2736399f8aaa6145c0c98e Mon Sep 17 00:00:00 2001
From: "heeju.joo" <heeju.joo@samsung.com>
Date: Thu, 22 Aug 2013 18:22:10 +0900
Subject: [PATCH] JQM Support hostKeyboard Enter key on button

---
 .../js/jquery.mobile.buttonMarkup.js               |   10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/libs/js/jquery-mobile-1.2.0/js/jquery.mobile.buttonMarkup.js b/libs/js/jquery-mobile-1.2.0/js/jquery.mobile.buttonMarkup.js
index 8bca1a4..c6ee880 100644
--- a/libs/js/jquery-mobile-1.2.0/js/jquery.mobile.buttonMarkup.js
+++ b/libs/js/jquery-mobile-1.2.0/js/jquery.mobile.buttonMarkup.js
@@ -262,7 +262,7 @@ var attachEvents = function() {
 	var hoverDelay = $.mobile.buttonMarkup.hoverDelay, hov, foc;
 
 	$.mobile.$document.bind( {
-		"vmousedown vmousecancel vmouseup vmouseover vmouseout focus blur scrollstart touchend touchcancel": function( event ) {
+		"vmousedown vmousecancel vmouseup vmouseover vmouseout focus blur scrollstart touchend touchcancel keypress": function( event ) {
 			var theme,
 				$btn = $( closestEnabledButton( event.target ) ),
 				isTouchEvent = event.originalEvent && /^touch/.test( event.originalEvent.type ),
@@ -299,6 +299,14 @@ var attachEvents = function() {
 					if ( foc ) {
 						clearTimeout( foc );
 					}
+				} else if ( evt === "keypress" ) {
+					if ( event.originalEvent.keyCode == 13 ) { // ENTER keyCode == 13
+						var parent_popup = $btn.parents(".ui-popup");
+						if ( parent_popup.length && ! parent_popup.parent().hasClass("ui-popup-active") ) {
+							return;
+						}
+						$btn.click();
+					}
 				}
 			}
 		},
-- 
1.7.9.5

